pkg/cmd/cli/cli_changefeed_create.go

```diff
@@ -0,0 +1,495 @@
+func (o *changefeedCommonOptions) strictDecodeConfig(component string, cfg *config.ReplicaConfig) error {
+	err := util.StrictDecodeFile(o.configFile, component, cfg)
+	if err != nil {
+		return err
+	}
+
+	_, err = filter.VerifyRules(cfg)
+
+	return err
+}
+
+// createChangefeedOptions defines common flags for the `cli changefeed crate` command.
+type createChangefeedOptions struct {
+	commonChangefeedOptions *changefeedCommonOptions
+
+	etcdClient *kv.CDCEtcdClient
+	pdClient   pd.Client
+
+	pdAddr     string
+	credential *security.Credential
+
+	changefeedID            string
+	disableGCSafePointCheck bool
+	startTs                 uint64
+}
+
+// newCreateChangefeedOptions creates new options for the `cli changefeed create` command.
+func newCreateChangefeedOptions(commonChangefeedOptions *changefeedCommonOptions) *createChangefeedOptions {
+	return &createChangefeedOptions{
+		commonChangefeedOptions: commonChangefeedOptions,
+	}
+}
+
+// addFlags receives a *cobra.Command reference and binds
+// flags related to template printing to it.
+func (o *createChangefeedOptions) addFlags(cmd *cobra.Command) {
+	if o == nil {
+		return
+	}
+
+	o.commonChangefeedOptions.addFlags(cmd)
+	cmd.PersistentFlags().StringVarP(&o.changefeedID, "changefeed-id", "c", "", "Replication task (changefeed) ID")
+	cmd.PersistentFlags().BoolVarP(&o.disableGCSafePointCheck, "disable-gc-check", "", false, "Disable GC safe point check")
+	cmd.PersistentFlags().Uint64Var(&o.startTs, "start-ts", 0, "Start ts of changefeed")
+}
+
+// complete adapts from the command line args to the data and client required.
+func (o *createChangefeedOptions) complete(ctx context.Context, f factory.Factory) error {
+	etcdClient, err := f.EtcdClient()
+	if err != nil {
+		return err
+	}
+
+	o.etcdClient = etcdClient
+
+	pdClient, err := f.PdClient()
+	if err != nil {
+		return err
+	}
+
+	o.pdClient = pdClient
+
+	o.pdAddr = f.GetPdAddr()
+	o.credential = f.GetCredential()
+
+	if o.startTs == 0 {
+		ts, logical, err := o.pdClient.GetTS(ctx)
+		if err != nil {
+			return err
+		}
+		o.startTs = oracle.ComposeTS(ts, logical)
+	}
+
+	return nil
+}
+
+// validate checks that the provided attach options are specified.
+func (o *createChangefeedOptions) validate(ctx context.Context) error {
+	if o.commonChangefeedOptions.sinkURI == "" {
+		return errors.New("Creating changefeed without a sink-uri")
+	}
+
+	if err := o.validateStartTs(ctx); err != nil {
+		return err
+	}
+
+	return o.validateTargetTs()
+}
+
+func (o *createChangefeedOptions) getInfo(ctx context.Context, cmd *cobra.Command) (*model.ChangeFeedInfo, error) {
+	_, captureInfos, err := o.etcdClient.GetCaptures(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	cdcClusterVer, err := version.GetTiCDCClusterVersion(captureInfos)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	cfg := config.GetDefaultReplicaConfig()
+
+	if !cdcClusterVer.ShouldEnableOldValueByDefault() {
```

Comment:
May be it is better that we keep the check about  `EnableOldValue` in one place.